lib/plugins/packages.js

```diff
@@ -0,0 +1,252 @@
+// core dependencies
+const path = require('path')
+
+// npm dependencies
+const fse = require('fs-extra')
+
+// local dependencies
+const { startPerformanceTimer, endPerformanceTimer } = require('../utils/performance')
+const { packageDir, projectDir } = require('../utils/paths')
+const { requestHttpsJson } = require('../utils/requestHttps')
+const { verboseLog } = require('../utils/verboseLogger')
+const knownPlugins = require(path.join(packageDir, 'known-plugins.json'))
+const projectPackage = require(path.join(projectDir, 'package.json'))
+const config = require('../config')
+const { getConfigForPackage } = require('../utils/requestHttps')
+
+let packageTrackerInterval
+
+const packagesCache = {}
+
+// This allows npm modules to act as if they are plugins by providing the plugin config for them
+const proxyPluginConfig = {
```

Comment:
This has been duplicated, we should keep one version (if it's needed in both places we should look up that one version in both places)

Reply:
Yes, I think the packagesCache was the original